simplesamlphp/config/acquia_config.php

```diff
@@ -0,0 +1,125 @@
+
+// Change these for your installation.
+$config['secretsalt'] = SamlSecrets::SECRET_SALT;
+$config['auth.adminpassword'] = SamlSecrets::AUTH_ADMIN_PASSWORD;
+
+// Protect the main SAML admin page.
+$config['admin.protectindexpage'] = true;
+
+/**
+ * Support SSL Redirects to SAML login pages.
+ *
+ * Uncomment the code following code block to set
+ * server port to 443 on HTTPS environment.
+ *
+ * This is a requirement in SimpleSAML when providing a redirect path.
+ *
+ * @link https://github.com/simplesamlphp/simplesamlphp/issues/450
+ *
+ */
+$_SERVER['SERVER_PORT'] = 443;
+$_SERVER['HTTPS'] = 'true';
+$protocol = 'https://';
+$port = ':' . $_SERVER['SERVER_PORT'];
+
+/**
+ * Support multi-site and single site installations at different base URLs.
+ *
+ * Overide $config['baseurlpath'] = "https://{yourdomain}/simplesaml/"
+ * to customize the default Acquia configuration.
+ */
+$config['baseurlpath'] = $protocol . $_SERVER['HTTP_HOST'] . $port . '/simplesaml/';
+
+/**
+ * Cookies No Cache.
+ *
+ * Allow users to be automatically logged in if they signed in via the same
+ * SAML provider on another site by uncommenting the setcookie line below.
+ *
+ * Warning: This has performance implications for anonymous users.
+ *
+ * @link https://docs.acquia.com/resource/using-simplesamlphp-acquia-cloud-site
+ */
+// setcookie('NO_CACHE', '1');.
+/**
+ * Generate Acquia session storage via hosting creds.json.
+ *
+ * Session sorage defaults using the database for the current request.
+ *
+ * @link https://docs.acquia.com/resource/using-simplesamlphp-acquia-cloud-site/#storing-session-information-using-the-acquia-cloud-sql-database
+ */
+
+if (!getenv('AH_SITE_ENVIRONMENT')) {
+  // Add / modify your local configuration here.
+  $config['store.type'] = 'sql';
+  $config['store.sql.dsn'] = sprintf('mysql:host=%s;port=%s;dbname=%s', '127.0.0.1', '', 'drupal');
+  $config['store.sql.username'] = 'drupal';
+  $config['store.sql.password'] = 'drupal';
+  $config['store.sql.prefix'] = 'simplesaml';
+  $config['certdir'] = "/var/www/simplesamlphp/cert/";
+  $config['metadatadir'] = "/var/www/simplesamlphp/metadata";
+  $config['baseurlpath'] = 'simplesaml/';
+  $config['loggingdir'] = '/var/www/simplesamlphp/log/';
+
+}
+elseif (getenv('AH_SITE_ENVIRONMENT')) {
+  // Set  ACE ad ACSF sites based on hosting database and site name.
+  $config['certdir'] = "/mnt/files/{$_ENV['AH_SITE_GROUP']}.{$_ENV['AH_SITE_ENVIRONMENT']}/saml-certificate/";
+  $config['metadatadir'] = "/mnt/www/html/{$_ENV['AH_SITE_GROUP']}.{$_ENV['AH_SITE_ENVIRONMENT']}/simplesamlphp/metadata";
+  $config['baseurlpath'] = 'simplesaml/';
+  // Setup basic logging.
+  $config['logging.handler'] = 'file';
+  $config['loggingdir'] = dirname(getenv('ACQUIA_HOSTING_DRUPAL_LOG'));
+  $config['logging.logfile'] = 'simplesamlphp-' . date('Ymd') . '.log';
+  $creds_json = file_get_contents('/var/www/site-php/' . $_ENV['AH_SITE_GROUP'] . '.' . $_ENV['AH_SITE_ENVIRONMENT'] . '/creds.json');
```

Comment:
What's the plan for ACSF?  Is it a post task, or what? (Per the docs this needs to be sites.json or something when running under ACSF)

Reply:
Agreed - It likely makes sense to do V1 for ACE and build on it for the ACSF conversion in a new PR.

Reply:
The plan, such as it is, is to sniff whether the code is running on ACE or ACSF and load the appropriate credentials file accordingly.